CMakeLists.txt

```diff
@@ -6,17 +6,40 @@ file(GLOB_RECURSE sources CONFIGURE_DEPENDS
   ${CMAKE_SOURCE_DIR}/src/*.cpp
   ${CMAKE_SOURCE_DIR}/src/data/models/odb-gen/*.cxx
 )
+list(REMOVE_ITEM sources ${CMAKE_SOURCE_DIR}/src/main.cpp)
```

Comment:
during the parsing of cpp files, you can change their search path to something like `${CMAKE_SOURCE_DIR}/src/*/*.cpp` so that you don't have to specifically delete main.cpp from the list